--- a/angr/state_plugins/plugin.py
+++ b/angr/state_plugins/plugin.py
@@ -20,7 +20,11 @@ class SimStatePlugin(object):
         """
         Sets a new state (for example, if the state has been branched)
         """
-        self.state = state._get_weakref()
+        from angr.state_plugins import SimStateHistory
+        if isinstance(self, SimStateHistory):
+            self.state = state._get_strongref()
+        else:
+            self.state = state._get_weakref()

     def set_strongref_state(self, state):
         pass
